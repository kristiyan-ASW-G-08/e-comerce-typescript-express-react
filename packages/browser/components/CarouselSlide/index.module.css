.blue {
  background-color: #2563eb;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='6' viewBox='0 0 6 6' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='%23dbeafe' fill-opacity='0.89' fill-rule='evenodd'%3E%3Cpath d='M5 0h1L0 6V5zM6 5v1H5z'/%3E%3C/g%3E%3C/svg%3E");
}

.red {
  background-color: #dc2626;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='6' viewBox='0 0 6 6' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='%23fee2e2' fill-opacity='0.4' fill-rule='evenodd'%3E%3Cpath d='M5 0h1L0 6V5zM6 5v1H5z'/%3E%3C/g%3E%3C/svg%3E");
}
.green {
  background-color: #16a34a;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='6' viewBox='0 0 6 6' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='%23dcfce7' fill-opacity='0.4' fill-rule='evenodd'%3E%3Cpath d='M5 0h1L0 6V5zM6 5v1H5z'/%3E%3C/g%3E%3C/svg%3E");
}

.grey {
  background-color: #475569;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='6' viewBox='0 0 6 6' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='%23f1f5f9' fill-opacity='0.4' fill-rule='evenodd'%3E%3Cpath d='M5 0h1L0 6V5zM6 5v1H5z'/%3E%3C/g%3E%3C/svg%3E");
}
