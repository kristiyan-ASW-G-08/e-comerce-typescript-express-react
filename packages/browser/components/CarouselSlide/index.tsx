import { Dispatch, FC, SetStateAction } from 'react';
import Image from 'next/image';
import image1 from 'assets/headphone-blue.png';
import styles from './index.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faDotCircle, faCircle } from '@fortawesome/free-solid-svg-icons';
import Link from 'next/link';

interface CarouselSlideProps {
  heading: string;
  content: string;
  buttonText: string;
  setCurrentSlide: Dispatch<SetStateAction<number>>;
  currentSlide: number;
  imageSrc: any;
  color: 'green' | 'red' | 'blue' | 'grey';
  href: string;
}
const CarouselSlide: FC<CarouselSlideProps> = ({
  heading,
  content,
  buttonText,
  setCurrentSlide,
  currentSlide,
  imageSrc,
  color,
  href,
}) => {
  const hoverClasses = {
    red: 'hover:bg-red-700',
    green: 'hover:bg-green-700',
    blue: 'hover:bg-blue-700',
  };
  return (
    <div
      key={color}
      className={`${styles[color]} animate-slide mx-5 h-60 md:h-96	w-full rounded-lg flex flex-col  justify-center  `}
    >
      <div className="flex justify-center">
        <div className="m-auto p-5">
          <h1 className="pb-3 font-bold text-2xl md:text-5xl text-neutral-50">
            {heading}
          </h1>
          <p className="pb-10 font-bold md:text-lg text-neutral-50">
            {content}
          </p>
          <Link href={href}>
            <a>
              <button
                className={`text-sm rounded-full border-2 py-2 px-7 text-neutral-50 font-bold ${hoverClasses[color]}`}
              >
                {buttonText}
              </button>
            </a>
          </Link>
        </div>
        <div className="md:h-72 md:w-72 h-48 w-48 relative m-auto">
          {' '}
          {imageSrc ? (
            <Image src={imageSrc} layout="fill" alt="" className="h-40" />
          ) : (
            ''
          )}
        </div>
      </div>
      <div className="flex justify-center mb-5">
        <button
          data-testid="button-0"
          onClick={() => setCurrentSlide(0)}
          className="px-3 text-neutral-50"
        >
          {' '}
          {currentSlide === 0 ? (
            <FontAwesomeIcon height={12} icon={faCircle} />
          ) : (
            <FontAwesomeIcon height={12} icon={faDotCircle} />
          )}
        </button>

        <button
          data-testid="button-1"
          onClick={() => setCurrentSlide(1)}
          className="px-3 text-neutral-50"
        >
          {' '}
          {currentSlide === 1 ? (
            <FontAwesomeIcon height={12} icon={faCircle} />
          ) : (
            <FontAwesomeIcon height={12} icon={faDotCircle} />
          )}
        </button>

        <button
          data-testid="button-2"
          onClick={() => setCurrentSlide(2)}
          className="px-3 text-neutral-50"
        >
          {' '}
          {currentSlide === 2 ? (
            <FontAwesomeIcon height={12} icon={faCircle} />
          ) : (
            <FontAwesomeIcon height={12} icon={faDotCircle} />
          )}
        </button>
      </div>
    </div>
  );
};

export default CarouselSlide;
