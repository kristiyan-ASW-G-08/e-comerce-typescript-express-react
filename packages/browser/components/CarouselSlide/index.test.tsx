import React, { Dispatch, SetStateAction } from 'react';
import { render, waitFor } from '@testing-library/react';
import CarouselSlide from '.';
import userEvent from '@testing-library/user-event';

describe('CarouselSlide', () => {
  afterEach(() => jest.resetAllMocks());
  afterAll(() => jest.restoreAllMocks());
  it('render ', async () => {
    // expect.assertions(2);
    const heading = 'Browse Our Promotions';
    const imageSrc = '';
    const cardNum = 'card1';
    const content = 'Wireless Connection With TV, Computer, Laptop..';
    const setCurrentSlide = jest.fn() as Dispatch<SetStateAction<number>>;
    const { container, getByTestId } = render(
      <CarouselSlide
        heading={heading}
        content={content}
        buttonText={'someText'}
        setCurrentSlide={setCurrentSlide}
        currentSlide={0}
        imageSrc={imageSrc}
        color={'blue'}
      />,
      {},
    );

    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
    const buttons = [0, 1, 2];
    buttons.forEach(buttonNum => {
      const button = getByTestId(`button-${buttonNum}`);
      userEvent.click(button);
      waitFor(() => {
        expect(setCurrentSlide).toHaveReturnedWith(buttonNum);
      });
    });
  });
});
