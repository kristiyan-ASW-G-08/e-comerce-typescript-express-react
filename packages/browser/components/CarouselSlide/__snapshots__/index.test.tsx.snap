// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSlide render  1`] = `
<div>
  <div
    class="undefined animate-slide mx-5 h-60 md:h-96	w-full rounded-lg flex flex-col  justify-center  "
  >
    <div
      class="flex justify-center"
    >
      <div
        class="m-auto p-5"
      >
        <h1
          class="pb-3 font-bold text-2xl md:text-5xl text-neutral-50"
        >
          Browse Our Promotions
        </h1>
        <p
          class="pb-10 font-bold md:text-lg text-neutral-50"
        >
          Wireless Connection With TV, Computer, Laptop..
        </p>
        <button
          class="text-sm rounded-full border-2 py-2 px-7 text-neutral-50 font-bold hover:bg-blue-900"
        >
          someText
        </button>
      </div>
      <div
        class="md:h-72 md:w-72 h-48 w-48 relative m-auto"
      >
         
      </div>
    </div>
    <div
      class="flex justify-center mb-5"
    >
      <button
        class="px-3 text-neutral-50"
        data-testid="button-0"
      >
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle "
          data-icon="circle"
          data-prefix="fas"
          focusable="false"
          height="12"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="px-3 text-neutral-50"
        data-testid="button-1"
      >
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-dot "
          data-icon="circle-dot"
          data-prefix="fas"
          focusable="false"
          height="12"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 352C309 352 352 309 352 256C352 202.1 309 160 256 160C202.1 160 160 202.1 160 256C160 309 202.1 352 256 352z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="px-3 text-neutral-50"
        data-testid="button-2"
      >
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-dot "
          data-icon="circle-dot"
          data-prefix="fas"
          focusable="false"
          height="12"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 352C309 352 352 309 352 256C352 202.1 309 160 256 160C202.1 160 160 202.1 160 256C160 309 202.1 352 256 352z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
